service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/ControlledAiNotebookHandler.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.workspace.service.resource.controlled.cloud.gcp.ainotebook;
+
+import bio.terra.workspace.db.DbSerDes;
+import bio.terra.workspace.db.model.DbResource;
+import bio.terra.workspace.service.resource.model.WsmResource;
+import bio.terra.workspace.service.resource.model.WsmResourceHandler;
+import bio.terra.workspace.service.workspace.GcpCloudContextService;
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ControlledAiNotebookHandler implements WsmResourceHandler {
+  private static ControlledAiNotebookHandler theHandler;
+  private final GcpCloudContextService gcpCloudContextService;
+
+  @Autowired
+  public ControlledAiNotebookHandler(GcpCloudContextService gcpCloudContextService) {
+    this.gcpCloudContextService = gcpCloudContextService;
+  }
+
+  public static ControlledAiNotebookHandler getHandler() {
+    return theHandler;
+  }
+
+  @PostConstruct
```

Comment:
i've never seen postConstruct before and it seems to be deprecated? https://levelup.gitconnected.com/stop-using-postconstruct-in-your-java-applications-2a66fb202cb8

Could you autoWire ControlledAiNoteBookHandler into ResourceDao instead of using getHandler?

Reply:
I could `@Autowire`, but that defeats my goal for new resources _not_ to have to touch `ResourceDao`.

AFAIK, `@PostConstruct` is not deprecated, but it must be used with care. In our system, we do not start taking requests or running any flights until all of the Spring objects have been built. The problems in using `@PostConstruct` are caused by a system in operation, making a race between usage and the execution of the `@PostConstruct`.

All `@PostConstruct` methods are done executing before Spring does injection. All of that is before our `StartupInitializer` is called.